src/JavaCodeGenerator.cs

```diff
@@ -688,7 +688,17 @@ private static Service ParseService(AutoRestCodeModel codeModel, JavaSettings se
 
             IEnumerable<XmlSequenceWrapper> xmlSequenceWrappers = ParseXmlSequenceWrappers(codeModel, settings);
 
-            IEnumerable<ServiceModel> models = ParseModels(codeModel, settings);
+            #region Parse Models
+            ServiceModels serviceModels = new ServiceModels();
+            IEnumerable<AutoRestCompositeType> autoRestModelTypes = codeModel.ModelTypes
+                .Union(codeModel.HeaderTypes)
+                .Where((AutoRestCompositeType autoRestModelType) => ShouldParseModelType(autoRestModelType, settings));
+            List<ServiceModel> models = new List<ServiceModel>();
```

Comment:
Consider dropping the region, then `models = autoRestModelTypes.Select(ParseModel).ToList()`

Reply:
I like the region, just because it conveys the boundaries of that parsing action. I didn't want to extract it into its own function because it's only used here.
I'll simplify the foreach() to a LINQ query.